### PR TITLE
refactor: introduce vCluster manager

### PR DESCRIPTION
**Pull request by FabianKramm**
**What issue type does this pull request address?** (keep at least one, remove the others) 
/kind enhancement

**What does this pull request do? Which issues does it resolve?** (use `resolves #<issue_number>` if possible) 
Resolves ENG-3491
Resolves ENG-3493

### Changes:
* New `vcluster use manager helm|platform` command
* Added platform functionality for `vcluster create`, `vcluster connect`, `vcluster list`, `vcluster pause`, `vcluster resume` & `vcluster delete`
* Renamed `vcluster import` -> `vcluster activate`
* Renamed `vcluster pro` -> `vcluster platform`
* Removed `vcluster pro generate admin-kube-config`
* Renamed `platform.apiKey` -> `platform.api`
* Moved cli functionality from `cmd/vclusterctl/cmd` to `pkg/cli`
* Make e2e tests reusable for vcluster-pro (move actual tests into packages, so they can be imported separately)
* Moved `pkg/config/legacyconfig` -> `config/legacyconfig`

### CODE REVIEW

**Comment by FabianKramm** [comment]
@heiko-braun yes, why shouldn't we?

File: pkg/cli/connect_platform.go [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+	kubeConfig, err := clihelper.GetProKubeConfig(contextOptions)
+	if err != nil {
+		return nil, err
+	}
+
+	// we want to use a service account token in the kube config
+	if cmd.ServiceAccount != "" {
+		// check if its enabled on the pro vcluster
+		if !vCluster.VirtualCluster.Status.VirtualCluster.ForwardToken {
+			return nil, fmt.Errorf("forward token is not enabled on the vCluster and hence you cannot authenticate with a service account token")
```
Comment (johannesfrey) [inline pkg/cli/connect_platform.go:145]:
```suggestion
			return nil, fmt.Errorf("forward token is not enabled on the virtual cluster and hence you cannot authenticate with a service account token")
```

File: pkg/cli/find/find.go [hunk 2]
```diff
@@ -65,6 +66,49 @@ func CurrentContext() (string, *clientcmdapi.Config, error) {
...
+	}
+
+	// match answer
+	for idx, s := range questionOptions {
+		if s == selectedVCluster {
+			return &platformVClusters[idx], nil
+		}
+	}
+
+	return nil, fmt.Errorf("unexpected error searching for selected vcluster")
```
Comment (johannesfrey) [inline pkg/cli/find/find.go:109]:
```suggestion
	return nil, fmt.Errorf("unexpected error searching for selected virtual cluster")
```

File: cmd/vclusterctl/cmd/disconnect.go [hunk 3]
```diff
@@ -66,8 +66,13 @@ func (cmd *DisconnectCmd) Run() error {
 	// get vcluster info from context
 	vClusterName, _, otherContext := find.VClusterFromContext(cmd.Context)
 	if vClusterName == "" {
-		return fmt.Errorf("current selected context \"%s\" is not a vcluster context. If you've used a custom context name you will need to switch manually using kubectl", otherContext)
+		// get vCluster platform info from context
+		vClusterName, _, otherContext = find.VClusterPlatformFromContext(cmd.Context)
+		if vClusterName == "" {
+			return fmt.Errorf("current selected context \"%s\" is not a vcluster context. If you've used a custom context name you will need to switch manually using kubectl", otherContext)
```
Comment (johannesfrey) [inline cmd/vclusterctl/cmd/disconnect.go:72]:
```suggestion
			return fmt.Errorf("current selected context \"%s\" is not a virtual cluster context. If you've used a custom context name you will need to switch manually using kubectl", otherContext)
```

File: cmd/vclusterctl/cmd/login.go [hunk 4]
```diff
@@ -7,52 +7,82 @@ import (
...
-		Use:   "login [VCLUSTER_PRO_HOST]",
-		Short: "Login to a vCluster.Pro instance",
+		Use:   "login [VCLUSTER_PLATFORM_HOST]",
+		Short: "Login to a vCluster platform instance",
 		Long:  description,
 		Args:  cobra.MaximumNArgs(1),
 		RunE: func(cobraCmd *cobra.Command, args []string) error {
 			if config.ShouldCheckForProFeatures() {
-				cmd.Log.Warnf("In order to use a Pro feature, please contact us at https://www.vcluster.com/pro-demo or downgrade by running `vcluster upgrade --version v0.19.5`")
+				log.GetInstance().Warnf("In order to use a Pro feature, please contact us at https://www.vcluster.com/pro-demo or downgrade by running `vcluster upgrade --version v0.19.5`")
```
Comment (johannesfrey) [inline cmd/vclusterctl/cmd/login.go:50]:
```suggestion
				log.GetInstance().Warnf("In order to use a Platform feature, please contact us at https://www.vcluster.com/pro-demo or downgrade by running `vcluster upgrade --version v0.19.5`")
```

File: cmd/vclusterctl/cmd/platform/access_key.go [hunk 5]
```diff
@@ -20,30 +20,33 @@ var (
...
 		log:         log.GetInstance(),
 	}
 
-	tokenCmd := &cobra.Command{
-		Use:   "token",
-		Short: "Token prints the access token to a vCluster.Pro instance",
+	accessKeyCmd := &cobra.Command{
+		Use:     "access-key",
+		Aliases: []string{"token"},
+		Short:   "Token prints the access token to a vCluster platform instance",
```
Comment (johannesfrey) [inline cmd/vclusterctl/cmd/platform/access_key.go:44]:
```suggestion
		Short:   "Access-key prints the access token to a vCluster platform instance",
```

File: pkg/cli/activate_platform.go [hunk 6]
```diff
@@ -133,7 +73,7 @@ func GetVClusterNamespace(ctx context.Context, context, name, namespace string,
 		return "", fmt.Errorf("please specify a name")
 	}
 
-	// list vclusters
+	// list vClusters
```
Comment (johannesfrey) [inline pkg/cli/activate_platform.go:76]:
```suggestion
	// list virtual clusters
```

File: cmd/vclusterctl/cmd/platform/pro.go [hunk 7]
```diff
@@ -1,26 +1,28 @@
...
+	"github.com/loft-sh/vcluster/pkg/cli/flags"
+	"github.com/loft-sh/vcluster/pkg/platform"
 	"github.com/spf13/cobra"
 )
 
 func NewProCmd(globalFlags *flags.GlobalFlags) (*cobra.Command, error) {
 	proCmd := &cobra.Command{
 		Use:   "pro",
-		Short: "vCluster.Pro subcommands",
+		Short: "vCluster pro subcommands",
```
Comment (johannesfrey) [inline cmd/vclusterctl/cmd/platform/pro.go:15]:
```suggestion
		Short: "vCluster platform subcommands",
```

File: cmd/vclusterctl/cmd/resume.go [hunk 8]
```diff
@@ -52,46 +50,25 @@ vcluster resume test --namespace test
...
-	// load the rest config
-	kubeConfig, err := vCluster.ClientFactory.ClientConfig()
-	if err != nil {
-		return fmt.Errorf("there is an error loading your current kube config (%w), please make sure you have access to a kubernetes cluster and the command `kubectl get namespaces` is working", err)
-	}
-
-	kubeClient, err := kubernetes.NewForConfig(kubeConfig)
-	if err != nil {
-		return err
+	// check if we should create a platform vCluster
```
Comment (johannesfrey) [inline cmd/vclusterctl/cmd/resume.go:68]:
```suggestion
	// check if we should resume a platform backed virtual cluster
```

File: cmd/vclusterctl/cmd/use/manager.go [hunk 9]
```diff
@@ -0,0 +1,74 @@
...
+	description := `########################################################
+################# vcluster use manager #################
+########################################################
+Either use helm or vCluster platform as the deployment method for managing vClusters.
+#######################################################
+	`
+
+	managerCmd := &cobra.Command{
+		Use:   "manager",
+		Short: "Switch managing method of vClusters between platform and helm",
```
Comment (johannesfrey) [inline cmd/vclusterctl/cmd/use/manager.go:34]:
```suggestion
		Short: "Switch managing method of virtual clusters between platform and helm",
```

File: cmd/vclusterctl/cmd/use/manager.go [hunk 10]
```diff
@@ -0,0 +1,74 @@
...
+func NewManagerCmd(globalFlags *flags.GlobalFlags) *cobra.Command {
+	cmd := &ManagerCmd{
+		GlobalFlags: globalFlags,
+		Log:         log.GetInstance(),
+	}
+
+	description := `########################################################
+################# vcluster use manager #################
+########################################################
+Either use helm or vCluster platform as the deployment method for managing vClusters.
```
Comment (johannesfrey) [inline cmd/vclusterctl/cmd/use/manager.go:28]:
```suggestion
Either use helm or vCluster platform as the deployment method for managing virtual clusters.
```

**Review by johannesfrey — COMMENTED** [review]
Mostly comments. I'm sure I missed cases of `vCluster vs. virtual cluster`. vCluster being the product/CLI vs. an actual virtual cluster that is running/created/updated etc.

File: cmd/vclusterctl/cmd/activate.go [hunk 11]
```diff
@@ -0,0 +1,54 @@
...
+
+Example:
+vcluster activate my-vcluster --cluster connected-cluster \
+--namespace vcluster-my-vcluster --project my-project --import-name my-vcluster
+#######################################################
+	`
+
+	importCmd := &cobra.Command{
+		Use:     "activate" + loftctlUtil.VClusterNameOnlyUseLine,
+		Aliases: []string{"import"},
```
[the remark this answers is not in the document]
Comment (FabianKramm) [reply cmd/vclusterctl/cmd/activate.go:33]: Yes, we want all three to be the same command, `vcluster import`, `vcluster activate` and `vcluster platform import`

**Review by johannesfrey — APPROVED** [review]

**Review by ThomasK33 — APPROVED** [review]

**Pull request merged**